tests/input_flows_helpers.py

```diff
@@ -365,3 +365,81 @@ def confirm_tx(
                     self.debug.press_left(wait=True)
                     self.debug.press_left(wait=True)
                 self.debug.press_middle()
+
+    def confirm_tx_staking(
+        self,
+        info: bool = False,
+    ) -> BRGeneratorType:
+        yield
+        TR.assert_equals_multiple(
+            self.debug.wait_layout().title(),
+            [
+                "ethereum__staking_stake",
+                "ethereum__staking_unstake",
+                "ethereum__staking_claim",
+            ],
+        )
+        TR.assert_equals_multiple(
+            self.debug.wait_layout().text_content(),
+            [
+                "ethereum__staking_stake_intro",
+                "ethereum__staking_unstake_intro",
+                "ethereum__staking_claim_intro",
+            ],
+        )
+        if self.debug.model == "T":
+            # confirm intro
+            if info:
+                self.debug.press_info(wait=True)
+                TR.assert_equals_multiple(
+                    self.debug.wait_layout().title(),
+                    [
+                        "ethereum__staking_stake_address",
+                        "ethereum__staking_claim_address",
+                    ],
+                )
+                self.debug.press_no(wait=True)
+            self.debug.press_yes()
+            yield
+
+            # confirm summary
+            if info:
+                self.debug.press_info(wait=True)
+                TR.assert_in(
+                    self.debug.wait_layout().text_content(), "ethereum__gas_limit"
+                )
+                TR.assert_in(
+                    self.debug.wait_layout().text_content(), "ethereum__gas_price"
+                )
+                self.debug.press_no(wait=True)
+            self.debug.press_yes()
+            yield
+        else:
+            # confirm intro
+            if info:
+                self.debug.press_right(wait=True)
+                TR.assert_equals_multiple(
+                    self.debug.wait_layout().title(),
+                    [
+                        "ethereum__staking_stake_address",
+                        "ethereum__staking_claim_address",
+                    ],
+                )
+                self.debug.press_left(wait=True)
+            self.debug.press_middle()
+            yield
+
+            # confirm summary
+            if info:
+                self.debug.press_right(wait=True)
+                TR.assert_in(
+                    self.debug.wait_layout().text_content(), "ethereum__gas_limit"
+                )
+                self.debug.press_right(wait=True)
+                TR.assert_in(
+                    self.debug.wait_layout().text_content(), "ethereum__gas_price"
+                )
+                self.debug.press_left(wait=True)
+                self.debug.press_left(wait=True)
+            self.debug.press_middle()
+            yield
```

Comment:
aside (not blocking the PR on it): this is absolutely horrible. I have zero idea what's going on here and if the test failed on me I wouldn't know how and why.

we MUST figure out a way to spell these test cases in some way that a human person can read

Reply:
I kinda agree and I was also getting used to it based on examples from other operations. I guess, one needs to know what the screens are supposed to look like.
The point here is to enter screens not necessarily seen by a user (info items) to validate that the staking address is there and that it's correctly `chunkified` (captured in `fixtures.json`)

Willing to discuss more readable ways how to do it.